### ISSUE TITLE
Unit test - RxQuery - Failed to write core dump

### ISSUE THREAD

**Issue by Makasky** [objectbox/objectbox-java#630]
## Issue Basics

- ObjectBox version : 2.2.0
- Reproducibility: often

### Description

Fatal error or AssertionError when I try to test a query with RxJava. It works fine when I don't use RxQuery

### Code

```
 class MovieDaoTest : AbstractObjectBoxTest() {

    val movies = listOf(
            Movie(1, "title", "overview", LocalDate.now(Clock.systemUTC())),
            Movie(2, "title", "overview", LocalDate.now(Clock.systemUTC())),
            Movie(3, "title", "overview", LocalDate.now(Clock.systemUTC()))
    )

    lateinit var box: Box<Movie>

    @Before
    override fun setUp() {
        super.setUp()
        box = store.boxFor()

        box.put(movies)
    }

    @Test
    fun getAll() {
        val query = box.query().build()
        RxQuery.observable(query)
                .test()
                .assertValue(movies)
    }
} 
```

### Logs & stackstraces

```
java.lang.AssertionError: Expected: [Movie(id=1, title=title, overview=overview, releaseDate=2018-12-19, createdAt=null, updatedAt=null), Movie(id=2, title=title, overview=overview, releaseDate=2018-12-19, createdAt=null, updatedAt=null), Movie(id=3, title=title, overview=overview, releaseDate=2018-12-19, createdAt=null, updatedAt=null)] (class: ArrayList), Actual: [[Movie(id=1, title=title, overview=overview, releaseDate=2018-12-19, createdAt=null, updatedAt=null), Movie(id=2, title=title, overview=overview, releaseDate=2018-12-19, createdAt=null, updatedAt=null), Movie(id=3, title=title, overview=overview, releaseDate=2018-12-19, createdAt=null, updatedAt=null)]] (latch = 1, values = 1, errors = 0, completions = 0)

	at io.reactivex.observers.BaseTestConsumer.fail(BaseTestConsumer.java:189)
	at io.reactivex.observers.BaseTestConsumer.assertValue(BaseTestConsumer.java:354)
	at com.app.somovie.db.MovieDaoTest.getAll(MovieDaoTest.kt:35)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMainV2.main(AppMainV2.java:131)

``` 
**OR**

```
 A fatal error has been detected by the Java Runtime Environment:

  SIGSEGV (0xb) at pc=0x000000010f343304, pid=9477, tid=0x000000000000a803

 JRE version: OpenJDK Runtime Environment (8.0_152-b06) (build 1.8.0_152-release-1136-b06)
 Java VM: OpenJDK 64-Bit Server VM (25.152-b06 mixed mode bsd-amd64 compressed oops)
 Problematic frame:
 C  [libobjectbox-jni-macos-x64.dylib+0x6304]  Java_io_objectbox_internal_DebugCursor_nativeSeekOrNext+0xcf4

 Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again

 An error report file with more information is saved as:
 /Users/fl/Documents/Projets/SoMovie/app/hs_err_pid9477.log

[INFO ] Creating query #1 for Movie with 1 condition(s)
[INFO ] Finding using query #1
```
[hs_err_pid9477.log](https://github.com/objectbox/objectbox-java/files/2696715/hs_err_pid9477.log)

**Comment by greenrobot-team** [comment]
Closing this issue due to inactivity. :zzz: Please re-open with more details or submit a new issue.
